### PR TITLE
USHIFT-1106: separate slow and fast verify checks

### PR DESCRIPTION
**Pull request by dhellmann**
Some of the verification checks take longer than others, so separate
them to provide a target developers can use locally without waiting
many minutes for tests to run.

/assign @copejon

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -115,22 +115,38 @@ etcd:
...
+# Default verify target for developers
+.PHONY: verify
+verify: verify-fast
 
+.PHONY: verify-ci
+verify-ci: verify-fast verify-slow
+
+# Fast verification checks that developers can/should run locally
+.PHONY: verify-fast
+verify-fast: verify-assets verify-sh verify-py verify-container licensecheck
```
[the remark this answers is not in the document]
Comment (dhellmann) [reply Makefile:127]: Yes for fmt and vet. The golangci-lint took a couple of minutes, but uses a cache so I could probably be talked into that.

I can rebase this PR on yours to add those, if you want, or we can do it the other way around.

[the remark this answers is not in the document]
Comment (copejon) [reply Makefile:127]: Running on an M1 mac, `golangci-lint run` takes ~10s to complete after the first run builds the cache. Even if the first run took 10m, that's still faster than waiting on CI to report linter errors (then fix them, then wait again for CI to check it).

[the remark this answers is not in the document]
Comment (pacevedom) [reply Makefile:127]: Verify job takes somewhere between 8 and 20 minutes in CI. Like Jon, I also had ~10s runs after the initial cache was built.

File: Makefile [hunk 2]
```diff
@@ -115,22 +115,35 @@ etcd:
...
+.PHONY: verify
+verify: verify-fast
 
+# Fast verification checks that developers can/should run locally
+.PHONY: verify-fast
+verify-fast: verify-go verify-assets verify-sh verify-py
+
+# Full verification checks that should run in CI
+.PHONY: verify-ci
+verify-ci: verify-fast verify-images verify-govulncheck licensecheck verify-container
```
Comment (copejon) [inline Makefile:128]: Let's also drop the vulnerability checks from CI until we have time to address the reported issues.  It should probably fall under the verify-go target anyhow.

Comment (dhellmann) [reply Makefile]: fixed

**Pull request merged**